lib/cached_network_image.dart

```diff
@@ -86,81 +90,177 @@ class CacheManager {
 
   CacheManager._();
 
-  SharedPreferences prefs;
-  Map<String, CacheObject> cacheData;
+  SharedPreferences _prefs;
+  Map<String, CacheObject> _cacheData;
+  DateTime lastCacheClean;
+
   static Object _lock = new Object();
 
   ///Shared preferences is used to keep track of the information about the files
   _init() async {
-    prefs = await SharedPreferences.getInstance();
+    _prefs = await SharedPreferences.getInstance();
 
     //get saved cache data from shared prefs
-    var jsonCacheString = prefs.getString("lib_cached_image_data");
-    cacheData = new Map();
+    var jsonCacheString = _prefs.getString("lib_cached_image_data");
+    _cacheData = new Map();
     if (jsonCacheString != null) {
       Map jsonCache = JSON.decode(jsonCacheString);
       jsonCache.forEach((key, data) {
-        cacheData[key] = new CacheObject.fromMap(data);
+        _cacheData[key] = new CacheObject.fromMap(key, data);
       });
     }
+
+    // Get data about when the last clean action has been performed
+    var cleanMillis = _prefs.getInt("lib_cached_image_data_last_clean");
```

Comment:
instead of putting comment you could extract method with the same name
`lastCacheClean = _getLastCleanTimestampFromPreferences();`

Reply:
Yes, I probably have to refactor a bit. I also want to extract the classes in multiple files.